### ISSUE TITLE
Give better error when Java version is wrong

### ISSUE THREAD

**Issue by sandain** [MinecraftServerControl/mscs#174]
According to @Roflicide, Java 8 is required for Minecraft versions >= 1.12 (see #173).  Are there any versions of Minecraft that can work with Java 6, and if so when did 7 become required?

Anyway, we have code in the script for testing Minecraft versions and we can extract the Java version with:
```
java -version 2>&1 | perl -ne 'if ($_ =~ /version\s"([\d\.\_]+)"/) { print $1 }'
```
All we need to do is setup a trap for invalid combinations and produce a better error message.

See #171 for the inspiration behind this.

**Comment by sandain** [comment]
According to [Mojang](https://help.mojang.com/customer/en/portal/articles/2636196-minecraft---java-6-and-7-support), up until 1.12 both Java 6 and 7 were supported.  Java 5 support was killed off with [version 1.6](https://minecraft.gamepedia.com/1.6.1) if we want to go that far back in time.

**Comment by zanix** [comment]
Nice, I was attempting to find this info as well.
Maybe check the Java version against the Minecraft version and display an error?
Also, a server can potentially have multiple versions of Java installed using `update-alternatives`, do we want to handle this or just keep using whatever `which java` returns?

**Comment by sandain** [comment]
Ohh goodness, I hadn't thought of that.  Debian based systems use `update-alternatives`, but I don't think that is universal.  The heuristics for this would be tricky.  Maybe we should just rely on `which java` and require the usage of the `mscs.defaults` or `mscs.properties` files for special cases.

**Comment by sandain** [comment]
Here is another thing to note: servers below Minecraft version 1.3.1 are not available at the update URL that we use.

**Comment by sandain** [comment]
Here are a couple of things to note: servers below Minecraft version 1.3.1 are not available at the update URL that we use, and a 1.3.1 server and client works just fine with Java 8.

**Comment by Roflicide** [comment]
I agree with @zanix. I think all we really need is something that checks the java version and tells the user the supported Minecraft versions based on the JRE they have (from `which java`) if they are running a JRE less than java 8. So something like:

If their running Java 5: Minecraft versions 1.3.1-1.6 supported
If their running Java 6 & 7: Minecraft versions 1.6.1-1.11.2 supported

We could add it so that they could change the `mscs.defaults` or `mscs.properties` files to specify JRE location, but I don't think it's really necessary. Maybe if someone specifically requests that then we could do it.

**Comment by sandain** [comment]
I think for the most part Java is backwards compatible with itself. So, if running Java 6 or 7, it would be: Minecraft versions 1.3.1-1.11.2 supported.

However, the script still supports even older versions of Minecraft than 1.3.1 -- it is just the update function that will fail, so in essence it would be the following:

    If running Java 5: Minecraft versions 1.3.1* through 1.5.2 supported.
    If running Java 6 or 7: Minecraft versions 1.3.1* through 1.11.2 supported.
    If running Java 8 or 9: All Minecraft versions are supported.

    [*] Versions older than 1.3.1 work with MSCS, but automatic download functionality for
    the server jar file is not available and will require manual installation.

The only way to modify the java location currently is with the `mscs-default-server-command` or `mscs-server-command` options in the `mscs.defaults` or `mscs.properties` files.  I agree with @Roflicide -- maybe we should add in `mscs-default-java` or `mscs-java` options to those same files to override `which java`.

**Comment by zanix** [comment]
I think we can leave it up to which java and override with a new setting.

**Comment by sandain** [comment]
I just sent in pull request #176.  I only tested to make sure that my Java 8 installation is still working without an error message, but I think that I'm trapping version combos that won't work.  Let me know what you think.

**Comment by sandain** [comment]
I closed pull request #176 because I never got around to fixing the issues identified.  Since using old Java versions is a security issue and even old versions of Minecraft should work on the latest Java version, this idea really doesn't provide much benefit.  Closing this issue.
